test_server.py

```diff
@@ -28,28 +32,53 @@ def helper(client, msg):
             return response
 
 
-def test_client1(make_client):
-    client = make_client
-    assert "HTTP/1.1 200 OK" in helper(client, "GET /path/templates/thing.html HTTP/1.1\r\nHOST: www.site.com\r\n\r\n")
+def parse_helper(request):
+    request = request.split("\r\n\r\n", 1)
+    req = request[0].split("\r\n")
+    for i, r in enumerate(req):
+        req[i] = r.split()
+    proto = req[0][0]
+    code = " ".join(req[0][1:])
+    body = request[-1]
+    headers = {}
+    for line in req[1:]:
+        headers[line[0].upper()] = line[1]
+    d = {'code': code, 'proto': proto, 'body': body}
+    d.update(headers)
+    return d
```

Comment:
Reading the names you use for your locals here, I expected this to be used to parse a response rather than a request.  Seeing the usage below appears to verify that.  Perhaps the parameter `request` is misnamed?